### ISSUE TITLE
Is there a possibility to install this kernel in arch?

### ISSUE THREAD

**Issue by guglielmobartelloni** [Grippentech/Asus-E200HA-Linux-Post-Install-Script#30]

**Comment by Grippy98** [comment]
It should technically work find. You could also try just "building it the arch way" from source. 

https://git.kernel.org/pub/scm/linux/kernel/git/tiwai/sound.git/log/?h=topic/asus-e100h-4.13

**Comment by liberodark** [comment]
and on manjaro for kernel 4.14 ?

**Comment by Grippy98** [comment]
Should be the same for that, although I don't believe Takashi Iwai ever did
a newer version than 4.13 at the time of writing.

**Comment by 7twin** [comment]
@Grippentech first of all, big thanks for actually putting so much work towards such a seemingly amazing but not very developed device!

I stumbled upon this repo from an antergos install (and the linux kernel comments linking back here) on the e200ha and wondered the same as OP and indeed Takashi did not release anything newer than 4.13 it seems, nor could I find anywhere else anything newer than that either.

The debian docs mention external soundcards would work atleast: https://wiki.debian.org/InstallingDebianOn/Asus/E200HA

Do you have any resources, that could help to use atleast an external soundcard with the e200ha? as it seems no troubleshooting makes any external soundcard appear.

I am sure that alternative option could atleast give a little bit of hope and come in handy, to the next people finding this repo.

Thanks again for such great work!

**Comment by Grippy98** [comment]
I realize it's super late now but basically any sound card that is confirmed to work with linux should work. 

Something like this: https://www.amazon.com/Sabrent-External-Adapter-Windows-AU-MMSA/dp/B00IRVQ0F8/ref=sr_1_2?ie=UTF8&qid=1521498765&sr=8-2&keywords=USB+soundcard#customerReviews

These should work without any extra drivers. 

In fact if you use something like this you might not need to flash a custom kernel at all.

**Comment by 7twin** [comment]
@Grippentech It's alright, I have since then figured it out, everything already plays well together on antergos (except for the sound of course), to get the sound via an external soundcard, I have found installing "cadence" works best.

**Comment by heikomat** [comment]
@7twin i was able to build a 4.16 kernel with cx2072x codec support, based on the fixes Takashi did for 4.13 and the infos from this repository.

I only tested it with debian and ubuntu, but maybe you can adapt this to arch? i'd love to know if this works on arch:

- [Kernel fork with cx2072x support](https://github.com/heikomat/linux/tree/cx2072x)
- [e200ha sound manual for debian & ubuntu](https://github.com/heikomat/linux/blob/cx2072x/cx2072x_fixes_and_manual/README.md)
- [Building the kernel manual for debian & ubuntu](https://github.com/heikomat/linux/blob/cx2072x/cx2072x_fixes_and_manual/building_the_kernel.md)

**Comment by 7twin** [comment]
@heikomat highly impressive that you got it to run on newer kernels, but I finally set up my laptop, that I am afraid adding the experimental kernel might break it. I might take some time off to sit down and try it some day, but it's not as much of a deal breaker to me anymore, since I can just use the usb soundcard.

**Comment by garchymede** [comment]
A bit late for commenting but I can confirm it runs fine on arch. I'm using a custom zen-kernel and i3 and it's really smooth. The only issue I had to resolve was the keyboard at boot since I fully encrypt the drives, and the sd cards that are not being recognized all the time, otherwise the only remaining issue is the updates of the kernels. The driver for the soundcard is a bit messy and needs some tweaking via alsamixer to open the right channels and build the correct bridges.

**Comment by 7twin** [comment]
@garchymede @heikomat I would be ready to try to refresh my laptop again, does this work with 4.17 kernel too? also @garchymede mentioned some issues with sd cards getting recognized, I have none on antergos currently, does this patch introduce those issues?

**Comment by heikomat** [comment]
@7twin
I merged 4.17 into my branch on my fork when it was tagged, and it seems to work just fine. I'm currently running Debian on 4.17 myself. Haven't tested the SD card, but I heard from a Linux mint user that uses the kernel, that the SD card reader seems to work too.

**Comment by 7twin** [comment]
@heikomat I seem to be missing all vital configurations and files in the configuring step here: https://github.com/heikomat/linux/blob/cx2072x/cx2072x_fixes_and_manual/building_the_kernel.md#configuring

- I don't have "Conexant CX2072 CODEC" nor "Baytrail and Cherrytrail with CX2072X codec"
- there's no /boot/config-* folder
- a lot of the prerequesite packages are not available in the repos

I couldn't find any fixes for those issues either, I wonder what @garchymede did to make it work.

**Comment by heikomat** [comment]
@7twin are you sure you are on the cx2072x branch of the repo (because of the missing menuconfig options)?

Also remember: these instructions are for Debian and work on my Ubuntu desktop. I don't know how to build the kernel for arch, as I haven't tried arch yet.

**Comment by 7twin** [comment]
@heikomat yeah I was back then pulling the wrong branch, I am now stuck in the after compilation, since I decided to do the same method described here: https://github.com/garchymede/archlinux_on_asus_E200HA to compille the kernel on my server and then just transfer whats needed to my laptop, but the issue is: it is 20GB big after compiling, so that won't ever fit onto the e200ha, I also fetched a fresh 4.17 kernel and applied the .config entries you did too, but it still results in 20GB, am I supposed to transfer only some files and then make modules_install, would you know that? or is there a way for me to use the source from your releases somehow?

**Comment by heikomat** [comment]
@7twin  Do you really need to transfer everything? When i build the kernel for debian, the resulting `.deb`-package is just about 50 MB

**Comment by 7twin** [comment]
@heikomat that's what I am trying to figure out, but can't find anything. I am right now compiling it with the config I got from my laptops /proc/config.gz and added the options/modules from your instructions, hoping it won't turn out to be 20GB again, since then I am clueless what else I could try and hope @garchymede would tell how he did it.

**Comment by 7twin** [comment]
@heikomat I finally got it compiled to just 2.9GB, I have also successfully booted off of it, but I face some issues as described here: https://github.com/garchymede/archlinux_on_asus_E200HA/issues/4 maybe if you encountered something like that on debian based systems, you could help?

**Comment by heikomat** [comment]
@7twin i haven't encountered such problems, but maybe you could dig through some logs to get info on what might be the problem. For a start, you could try this: https://github.com/heikomat/linux/blob/cx2072x/cx2072x_fixes_and_manual/README.md#getting-a-detailed-pulseaudio-log-for-when-debugging-is-necessary

**Comment by 7twin** [comment]
@heikomat after wiping my laptop and reinstalling it all, it works just fine! if I have the time I will write instructions for arch - based on your sources, in case you would ever stop maintaining it, do you just merge from linux:master and add the config? Thanks a lot for your work.

**Comment by heikomat** [comment]
@7twin sounds good. Yes, I do just merge from master on a new kernel release, update the config and rebuild the kernel.

**Comment by 7twin** [comment]
@heikomat awesome, thanks!

**Comment by heikomat** [comment]
@7twin I just added 4.18-rc6 with cx2072x to my fork. This time around, a simple merge wasn't enough.

There were no merge conflicts, but the kernel moved from `codecs` to `components` (whatever that means) (see [the last commits of soc.h](https://github.com/torvalds/linux/commits/master/include/sound/soc.h)).

I checked the changes from 4.17 to 4.18 in other codecs, and adjusted the cx2072x codec accordingly (see [here](https://github.com/heikomat/linux/commit/cd30431ed210bcdd7abd71fdb91f9b5b278e78be)). With these changes, sound is working with 4.18-rc6 just like it did with 4.17.

**Comment by 7twin** [comment]
@heikomat great work and thanks for keeping me updated!

**Comment by ErikHacklin** [comment]
Hello all!
Been fiddling with Arch Linux for a month or so now and the arch user repository (AUR) currently has a kernel available (linux-cx2072x) which works nicely. This kernel on the site links to heikomat's github page. Instructions for installing anything from AUR can be found from the Arch Wiki
